### ISSUE TITLE
ward sometimes changes content of strings

### ISSUE THREAD

**Issue by fingolfin** [gap-system/ward#32]
Running `testinstall` in  HPC-GAP on the `hpc-merge` branch produces, among other things, this diff:
```
########> Diff in /Users/mhorn/Projekte/GAP/gap.hpc-merge/build/hpcgap/../../hpcgap/tst/testinstall/trans.tst:1680
# Input is:
CYCLES_TRANS_LIST(IdentityTransformation, ());
# Expected output:
Error, CYCLES_TRANS_LIST: the second argument must be a list (not a transforma\
tion (small))
# But found:
Error, CYCLES_TRANS_LIST: the second argument must be a ReadGuard(list) (not a\
 transformation (small))
```

This looks as if ward got confused and modified an occurrence of `list` inside a string. Note that `list` is also the name of an argument to `FuncCYCLES_TRANS_LIST`):
```C
Obj FuncCYCLES_TRANS_LIST (Obj self, Obj f, Obj list) {
  UInt2   *ptf2;
  UInt4   *seen, *ptf4;
  UInt    deg, i, j, pt, nr;
  Obj     out, comp, list_i;

  if (!IS_TRANS(f)) {
    ErrorQuit("CYCLES_TRANS_LIST: the first argument must be a "
              "transformation (not a %s)", (Int) TNAM_OBJ(f), 0L);
  } else if (!IS_LIST(list)) {
    ErrorQuit("CYCLES_TRANS_LIST: the second argument must be a "
              "list (not a %s)", (Int) TNAM_OBJ(f), 0L);
  }
....
```

One guess is that the parser gets confused by the C convention that adjacent string literals get silently merged. Indeed, if I modify the code to merge the two adjacent strings, the test diff goes away.

**Comment by rbehrends** [comment]
The C parser is equipped to deal with this:

```lua
local string_constant = string_literal * (sp * string_literal)^0
```

The rewriting part uses a simplified parser that *may* get tripped up by this, but the analyzer should not even generate a rewrite for this line and the rewriter doesn't rename variables: it only adds guards. This is most peculiar.

**Comment by fingolfin** [comment]
For completeness, this is the output produced by ward:
```C
Obj FuncCYCLES_TRANS_LIST (Obj self, Obj f, Obj list) {
  UInt2   *ptf2;
  UInt4   *seen, *ptf4;
  UInt    deg, i, j, pt, nr;
  Obj     out, comp, list_i;

  ReadGuard(f);
#line 2894 "../../src/trans.c"
  if (!IS_TRANS(f)) {
    ErrorQuit("CYCLES_TRANS_LIST: the first argument must be a "
              "transformation (not a %s)", (Int) TNAM_OBJ(f), 0L);
  } else if (!IS_LIST(ReadGuard(list))) {
    ErrorQuit("CYCLES_TRANS_LIST: the second argument must be a "
              "ReadGuard(list) (not a %s)", (Int) TNAM_OBJ(f), 0L);
  }
...
```

**Comment by fingolfin** [comment]
Resolved by PR #34
